packages/happy-dom/src/nodes/html-table-section-element/HTMLTableSectionElement.ts

```diff
@@ -1,7 +1,88 @@
+import QuerySelector from '../../query-selector/QuerySelector.js';
+import * as PropertySymbol from '../../PropertySymbol.js';
 import HTMLElement from '../html-element/HTMLElement.js';
+import HTMLTableRowElement from '../html-table-row-element/HTMLTableRowElement.js';
+import DOMExceptionNameEnum from '../../exception/DOMExceptionNameEnum.js';
 /**
  * HTMLTableSectionElement
  *
  * @see https://developer.mozilla.org/en-US/docs/Web/API/HTMLTableSectionElement
  */
-export default class HTMLTableSectionElement extends HTMLElement {}
+export default class HTMLTableSectionElement extends HTMLElement {
+	/**
+	 * Returns an HTMLTableRowElement representing a new row of the table. It inserts it in the rows collection immediately before the <tr> element at the given index position. If the index is -1, the new row is appended to the collection. If the index is smaller than -1 or greater than the number of rows in the collection, a DOMException with the value IndexSizeError is raised.
+	 *
+	 * @param [index] Index.
+	 * @returns Row.
+	 */
+	public insertRow(index: number = -1): HTMLTableRowElement {
+		if (typeof index !== 'number') {
```

Comment:
I'm open to ideas on how to minimize duplication across the HTMLTableElement and HTMLTableSectionElement implementations. Some challenges:
* The error names are slightly different to reflect the name of the method it is called on
* Making sure the utility has the right "this"
* Different expectations on insert row since the HtmlTableElement must first find the tbody to add the row to

Reply:
I think it is fine with some duplication. Using a utility increases complexity and decreases readability to some extent.